src/models/game_piece.ts

```diff
@@ -51,9 +51,23 @@ class GamePiece extends Model<InferAttributes<GamePiece>, InferCreationAttribute
     return await (await this.playerUnit()).unit();
   }
 
+  // Returns the number of this game piece in the game, starting from 1
+  async gamePieceNumber(): Promise<number> {
+    const gamePieces = await Models.GamePiece.findAll({
+      where: {
+        gameId: this.gameId,
+      },
+      order: [
+        ['playerUnitId', 'ASC'], // hopefully this never changes?
```

Comment:
A GamePiece should definitely never have its playerUnitId change within a game, so we should be good here. Further, no two GamePieces within the same Game should ever be able to share a playerUnitId (i.e. you can't use "Bob the Archer" more than once in the same game), so I think this should work.